<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />
  <arg name="LOGGER_DEBUG" default="$(optenv LOGGER_DEBUG false)" />

  <env name="ROSCONSOLE_CONFIG_FILE" if="$(eval arg('LOGGER_DEBUG'))" value="$(find octomap_tools)/config/debug_verbosity.yaml" />

    <!-- other args -->
  <arg name="debug" default="false" />
  <arg name="standalone" default="true" />
  <arg name="custom_config" default="" />

  <arg name="map_path" default="$(env HOME)/maps" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="octomap_saver" args="$(arg nodelet) octomap_tools/OctomapSaver $(arg nodelet_manager)" output="screen">

      <rosparam file="$(find octomap_tools)/config/octomap_saver.yaml" command="load" />

      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="map_path" type="string" value="$(arg map_path)" />

      <!-- topics in -->
      <remap from="~octomap_in" to="octomap_full" />

    </node>

  </group>

</launch>
